idmclient/client.go

```diff
@@ -79,36 +79,46 @@ func (c *basicAuthClient) DoWithBody(req *http.Request, r io.ReadSeeker) (*http.
 // httpbakey.Client.VisitWebPage to perform an OAuth login interaction.
 func UbuntuSSOOAuthVisitWebPage(client *http.Client, tok *usso.SSOData) func(u *url.URL) error {
 	return func(u *url.URL) error {
-		return ussoOAuthVisit(client, tok, u)
+		lm, err := LoginMethods(client, u)
```

Comment:
After some thought, I'm wondering if this function actually belongs inside idmclient. ISTM that LoginMethods is exposed from idmclient so that calling packages can decide what login methods are available and call them when appropriate.

This particular function is only useful AFAICS if USSO is the *only* login method you want to support, which seems very specialised for the idmclient package.

And the distinction between UbuntuSSOOAuthVisit and UbuntuSSOOAuthVisitWebPage is really unclear.

So I'd rename UbuntuSSOOAuthVisit to UbuntuSSOOAuthVisitWebPage and move the current UbuntuSSOOAuthVisitWebPage out.